src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java

```diff
@@ -328,12 +316,26 @@ public void onEntityPortal(EntityPortalEvent event) {
             return;
         }
 
+        // Some shortcuts for later
+        Entity entity = event.getEntity();
+
+        Location fromLocation = this.locationManipulation.getBlockLocation(event.getFrom());
+        Location originalToLocation = this.locationManipulation.getBlockLocation(event.getTo());
+
+        World fromWorld = fromLocation.getWorld();
+        World originalToWorld = originalToLocation.getWorld();
+
+        if (fromWorld == null || originalToWorld == null) {
+            Logging.warning("from/to world is null in EntityPortalEvent for %s", entity.getName());
+            return;
+        }
+
         PortalType type;
-        if (originalTo.getWorld().getEnvironment() == World.Environment.NETHER
-                || (currentLocation.getWorld().getEnvironment() == World.Environment.NETHER && originalTo.getWorld().getEnvironment() == World.Environment.NORMAL)) {
+        if (fromWorld.getEnvironment() == World.Environment.NETHER
+                || (fromWorld.getEnvironment() == World.Environment.NETHER && originalToWorld.getEnvironment() == World.Environment.NORMAL)) {
             type = PortalType.NETHER;
-        } else if (originalTo.getWorld().getEnvironment() == World.Environment.THE_END
-                || (currentLocation.getWorld().getEnvironment() == World.Environment.THE_END && originalTo.getWorld().getEnvironment() == World.Environment.NORMAL)) {
+        } else if (fromWorld.getEnvironment() == World.Environment.THE_END
+                || (fromWorld.getEnvironment() == World.Environment.THE_END && originalToWorld.getEnvironment() == World.Environment.NORMAL)) {
```

Comment:
This seems wrong. The original checks check if the to world is nether/end OR the from world is nether/end and to world is normal.

Reply:
AH